### PR TITLE
Triangle patterns

### PR DESCRIPTION
**Pull request by Arens404**

### CODE REVIEW

File: src/simpleTrianglePatterns/simpleTrianglePattern.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package simpleTrianglePatterns;
+
+import java.util.Scanner;
+public class simpleTrianglePattern {
+    public static void main(String[] args){
+        Scanner scan = new Scanner(System.in);
+        System.out.print("Enter a number:");
+        try{
```
Comment (Chezo) [inline src/simpleTrianglePatterns/simpleTrianglePattern.java:9]: Are you sure, that all of these stuff will be in the try block?

File: src/simpleTrianglePatterns/simpleTrianglePattern.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package simpleTrianglePatterns;
+
+import java.util.Scanner;
+public class simpleTrianglePattern {
+    public static void main(String[] args){
+        Scanner scan = new Scanner(System.in);
```
Comment (Chezo) [inline src/simpleTrianglePatterns/simpleTrianglePattern.java]: Is it a meaningful name?

File: src/simpleTrianglePatterns/simpleTrianglePattern.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        try{
+            int n = scan.nextInt();
+            for(int i = 1; i <= n; i++){
+                for(int j = 1; j <= i; j++){
+                    System.out.print("*");
+                }
+                System.out.println();
+            }
+        } catch (Exception e) {
+            System.out.println("Wrong format! Please use numbers to enter the input!");
```
Comment (Chezo) [inline src/simpleTrianglePatterns/simpleTrianglePattern.java:17]: If i were you, I would have just written that "Wrong format! Please use numbers" but its just personal preference

**Review by Chezo — CHANGES_REQUESTED** [review]
Please consider to edit the commented areas

**Review by Arens404 — COMMENTED** [review]
Done!

File: src/simpleTrianglePatterns/simpleTrianglePattern2.java [hunk 4]
```diff
@@ -4,9 +4,9 @@
 
 public class simpleTrianglePattern2 {
     public static void main(String[] args){
-        Scanner scan = new Scanner(System.in);
-        System.out.print("Enter a number:");
         try{
+            Scanner scan = new Scanner(System.in);
```
Comment (Chezo) [inline src/simpleTrianglePatterns/simpleTrianglePattern2.java]: Well, just see my previous comment.

File: src/simpleTrianglePatterns/simpleTrianglePattern2.java [hunk 5]
```diff
@@ -4,9 +4,9 @@
...
+            System.out.print("Enter a number:");
             int n = scan.nextInt();
```
Comment (Chezo) [inline src/simpleTrianglePatterns/simpleTrianglePattern2.java]: is n a meaningful name?

File: src/simpleTrianglePatterns/simpleTrianglePattern2.java [hunk 6]
```diff
@@ -4,9 +4,9 @@
 
 public class simpleTrianglePattern2 {
     public static void main(String[] args){
-        Scanner scan = new Scanner(System.in);
```
Comment (Chezo) [inline src/simpleTrianglePatterns/simpleTrianglePattern2.java]: Is scan a meaningful name?

**Review by Chezo — CHANGES_REQUESTED** [review]

File: src/simpleTrianglePatterns/simpleTrianglePattern2.java (hunk 6 above)
Comment (Arens404) [reply src/simpleTrianglePatterns/simpleTrianglePattern2.java]: yes.

Comment (Chezo) [reply src/simpleTrianglePatterns/simpleTrianglePattern2.java]: Java fields must be nouns, like you cannot describe a state with a verb, and object fields are states.

File: src/simpleTrianglePatterns/simpleTrianglePattern.java [hunk 7]
```diff
@@ -3,10 +3,10 @@
 import java.util.Scanner;
 public class simpleTrianglePattern {
     public static void main(String[] args){
-        Scanner scan = new Scanner(System.in);
+        Scanner userInputScanner = new Scanner(System.in);
         System.out.print("Enter a number:");
         try{
-            int n = scan.nextInt();
+            int userInputScanner = scan.nextInt();
```
Comment (Chezo) [inline src/simpleTrianglePatterns/simpleTrianglePattern.java]: How can be an int a scanner? The int is just a primitive variable type which can store whole numbers (integers) Its not meaningful its misleading.

File: src/simpleTrianglePatterns/simpleTrianglePattern2.java [hunk 8]
```diff
@@ -4,10 +4,10 @@
...
+        Scanner userInputScanner = new Scanner(System.in);
         System.out.print("Enter a number:");
         try{
-            int n = scan.nextInt();
+            int n = userInputScanner.nextInt();
```
Comment (Chezo) [inline src/simpleTrianglePatterns/simpleTrianglePattern2.java]: its "n" again? :D

**Review by Chezo — CHANGES_REQUESTED** [review]

File: src/simpleTrianglePatterns/simpleTrianglePattern2.java (hunk 8 above)
Comment (Arens404) [reply src/simpleTrianglePatterns/simpleTrianglePattern2.java]: hehe :D

File: src/simpleTrianglePatterns/simpleTrianglePattern2.java [hunk 9]
```diff
@@ -7,12 +7,12 @@ public static void main(String[] args){
         Scanner userInputScanner = new Scanner(System.in);
         System.out.print("Enter a number:");
         try{
-            int n = userInputScanner.nextInt();
+            int number = userInputScanner.nextInt();
```
Comment (Chezo) [inline src/simpleTrianglePatterns/simpleTrianglePattern2.java]: Whats the difference between this number and an another number? Its still not meaningful.

File: src/simpleTrianglePatterns/simpleTrianglePattern.java [hunk 10]
```diff
@@ -6,11 +6,11 @@ public static void main(String[] args){
         Scanner userInputScanner = new Scanner(System.in);
         System.out.print("Enter a number:");
         try{
-            int userInputScanner = scan.nextInt();
+            int number = userInputScanner.nextInt();
         } catch (Exception e) {
```
Comment (Chezo) [inline src/simpleTrianglePatterns/simpleTrianglePattern.java]: How Exceptions works in java? What's the point with the inheritance? How catch works with exceptions? I'm sure this catch case just too broad. What if you got a runtime exception and the error message still say there's a wrong input?

**Review by Chezo — CHANGES_REQUESTED** [review]

File: src/simpleTrianglePatterns/simpleTrianglePattern.java (hunk 10 above)
Comment (Arens404) [reply src/simpleTrianglePatterns/simpleTrianglePattern.java]: bruhhhh

File: src/simpleTrianglePatterns/simpleTrianglePattern2.java (hunk 9 above)
Comment (Chezo) [reply src/simpleTrianglePatterns/simpleTrianglePattern2.java]: same

File: src/simpleTrianglePatterns/simpleTrianglePattern2.java [hunk 11]
```diff
@@ -8,8 +8,10 @@ public static void main(String[] args){
         System.out.print("Enter a number:");
         try{
             int number = userInputScanner.nextInt();
-        } catch (Exception e) {
+        } catch (WrongUserInputException wuie) {
```
Comment (Chezo) [inline src/simpleTrianglePatterns/simpleTrianglePattern2.java]: Own exception class not bad but this code will crash at a known case. also @cleanCode - meaningful names? e-wuie-rte???

**Review by Chezo — CHANGES_REQUESTED** [review]

File: src/simpleTrianglePatterns/simpleTrianglePattern2.java (hunk 9 above)
Comment (Chezo) [reply src/simpleTrianglePatterns/simpleTrianglePattern2.java]: same

File: src/simpleTrianglePatterns/simpleTrianglePattern2.java [hunk 12]
```diff
@@ -8,19 +8,19 @@ public static void main(String[] args){
...
+                for(int j = i; j <= number; j++){
+                    System.out.print(" ");
+                }
+                for(int j = 1;j <= i; j++){
+                    System.out.print("*");
+                }
+                System.out.println();
             }
-            System.out.println();
+        } catch (RuntimeException ex2){
```
Comment (Chezo) [inline src/simpleTrianglePatterns/simpleTrianglePattern2.java]: meaningful names? ex2?? ex1???? there's a case which obvious make the program crash, need to catch the exception! The IDE might be display it.

File: src/simpleTrianglePatterns/simpleTrianglePattern2.java [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+                for(int j = i; j <= number; j++){
+                    System.out.print(" ");
+                }
+                for(int j = 1;j <= i; j++){
+                    System.out.print("*");
+                }
+                System.out.println();
+            }
+        } catch (RuntimeException ex2){
+        System.out.println("Time out!");
```
Comment (Chezo) [inline src/simpleTrianglePatterns/simpleTrianglePattern2.java]: what kind of timeout in runtime exception?

**Review by Chezo — CHANGES_REQUESTED** [review]

File: src/simpleTrianglePatterns/simpleTrianglePattern2.java [hunk 14]
```diff
@@ -7,19 +7,19 @@ public static void main(String[] args){
...
+            for(int i = 1; i <= UserInputNumber; i++){
+                for(int j = i; j <= UserInputNumber; j++){
                     System.out.print(" ");
                 }
                 for(int j = 1;j <= i; j++){
                     System.out.print("*");
                 }
                 System.out.println();
             }
-        } catch (RuntimeException ex2){
```
Comment (Chezo) [inline src/simpleTrianglePatterns/simpleTrianglePattern2.java]: Lets check how exceptions works, why is it a bad thing that you uses Exception to write out the wrong exception? also, runtime exception not needed here. Here will be only 1 very specific exception which is thrown when the input is not a number!

File: src/simpleTrianglePatterns/simpleTrianglePattern2.java (hunk 13 above)
Comment (Chezo) [reply src/simpleTrianglePatterns/simpleTrianglePattern2.java]: exception2 still not meaningful.

File: src/simpleTrianglePatterns/simpleTrianglePattern2.java [hunk 15]
```diff
@@ -7,19 +7,19 @@ public static void main(String[] args){
...
+        } catch (RuntimeException exception2){
         System.out.println("Time out!");
-        } catch (Exception ex1) {
+        } catch (Exception exception1) {
```
Comment (Chezo) [inline src/simpleTrianglePatterns/simpleTrianglePattern2.java]: exception1 not meaningfull at all!

File: src/simpleTrianglePatterns/simpleTrianglePattern2.java [hunk 16]
```diff
@@ -7,19 +7,19 @@ public static void main(String[] args){
...
                     System.out.print(" ");
                 }
                 for(int j = 1;j <= i; j++){
                     System.out.print("*");
                 }
                 System.out.println();
             }
-        } catch (RuntimeException ex2){
+        } catch (RuntimeException exception2){
         System.out.println("Time out!");
```
Comment (Chezo) [inline src/simpleTrianglePatterns/simpleTrianglePattern2.java]: what kind of timeout ?

**Review by Chezo — CHANGES_REQUESTED** [review]

File: src/simpleTrianglePatterns/simpleTrianglePattern2.java [hunk 17]
```diff
@@ -7,19 +7,17 @@ public static void main(String[] args){
...
                 }
                 for(int j = 1;j <= i; j++){
                     System.out.print("*");
                 }
                 System.out.println();
             }
-        } catch (RuntimeException exception2){
-        System.out.println("Time out!");
-        } catch (Exception exception1) {
+        } catch (Exception WrongFormatException) {
```
Comment (Chezo) [inline src/simpleTrianglePatterns/simpleTrianglePattern2.java]: The exception name is still misleading and the type is still too broad

**Review by Chezo — CHANGES_REQUESTED** [review]

File: src/simpleTrianglePatterns/simpleTrianglePattern2.java [hunk 18]
```diff
@@ -17,8 +17,8 @@ public static void main(String[] args){
                 }
                 System.out.println();
             }
-        } catch (Exception WrongFormatException) {
-        System.out.println("Wrong format! Please use numbers to enter the input!");
+        } catch (Exception InputMismatchException) {
+        System.out.println("Wrong format! Please use numbers!");
```
Comment (Chezo) [inline src/simpleTrianglePatterns/simpleTrianglePattern2.java]: Still, type too broad, name mislead

**Review by Chezo — CHANGES_REQUESTED** [review]

File: src/simpleTrianglePatterns/simpleTrianglePattern2.java (hunk 18 above)
Comment (Chezo) [reply src/simpleTrianglePatterns/simpleTrianglePattern2.java]: Still, type too broad, name mislead

**Review by Chezo — CHANGES_REQUESTED** [review]

File: src/simpleTrianglePatterns/simpleTrianglePattern2.java (hunk 18 above)
Comment (Chezo) [reply src/simpleTrianglePatterns/simpleTrianglePattern2.java]: catch (NumberFormatException numberFormatException) {

File: src/simpleTrianglePatterns/simpleTrianglePattern2.java [hunk 19]
```diff
@@ -0,0 +1,24 @@
...
+            for(int i = 1; i <= userInputNumber; i++){
+                for(int j = i; j <= userInputNumber; j++){
+                    System.out.print(" ");
+                }
+                for(int j = 1;j <= i; j++){
+                    System.out.print("*");
+                }
+                System.out.println();
+            }
+        } catch (Exception InputMismatchException) {
```
Comment (Chezo) [inline src/simpleTrianglePatterns/simpleTrianglePattern2.java]: } catch (InputMismatchException inputMismatchException) {

**Review by Chezo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**